tests/v2/validation/provisioning/rke2/cert_rotation_test.go

```diff
@@ -0,0 +1,173 @@
+package rke2
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	rkev1 "github.com/rancher/rancher/pkg/apis/rke.cattle.io/v1"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/extensions/cloudcredentials"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/extensions/machinepools"
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/rancher/rancher/tests/framework/pkg/wait"
+	"github.com/rancher/rancher/tests/integration/pkg/defaults"
+	"github.com/rancher/rancher/tests/v2/validation/provisioning"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+type V2ProvCertRotationTestSuite struct {
+	suite.Suite
+	session     *session.Session
+	client      *rancher.Client
+	config      *provisioning.Config
+	clusterName string
+	namespace   string
+}
+
+func (r *V2ProvCertRotationTestSuite) TearDownSuite() {
+	r.session.Cleanup()
+}
+
+func (r *V2ProvCertRotationTestSuite) SetupSuite() {
+	testSession := session.NewSession(r.T())
+	r.session = testSession
+
+	r.config = new(provisioning.Config)
+	config.LoadConfig(provisioning.ConfigurationFileKey, r.config)
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(r.T(), err)
+
+	r.client = client
+
+	r.clusterName = r.client.RancherConfig.ClusterName
+	r.namespace = r.client.RancherConfig.ClusterName
+}
+
+func (r *V2ProvCertRotationTestSuite) testCertRotation(provider Provider, kubeVersion string, nodesAndRoles []machinepools.NodeRoles, credential *cloudcredentials.CloudCredential) {
+	name := fmt.Sprintf("Provider_%s/Kubernetes_Version_%s/Nodes_%v", provider.Name, kubeVersion, nodesAndRoles)
+	r.Run(name, func() {
+		r.Run("initial", func() {
+			testSession := session.NewSession(r.T())
```

Comment:
I do agree with you Jake, as the refactor is out of scope for this PR, but I am curious why there is a need to have a nested r.Run here. I completely missed it in my initial review and now I'm curious.

Reply:
Ah that was because I copied this implementation from my encryption key rotation draft PR, where I had two clients, some initial setup that needed to be performed, creating 10k secrets and then rotating encryption keys a second time on a different user. Not really applicable here but it's a result of copy-paste.

Reply:
Also @annablender right now abstracting the underlying cluster these tests are running on requires a lot of manual setup, so there isn't really a good way to separate this out (or a need to until another test does so)

Reply:
OK makes sense. I think we can fix this in another PR, and I'm fine with this going in as it is. Since this is suppose to be a back port PR anyways.

Reply:
Just realized this is the backport lol.

@jakefhyde Sure, makes sense. I'm wondering if there would be a way to abstract some of that setup into the Setup() function or a helper function, so the test looks cleaner. Because the test itself is actually just 2 lines of code. I'm fine with possibly addressing that in another PR